packages/blade/src/components/Dropdown/useDropdown.ts

```diff
@@ -258,15 +279,32 @@ const useDropdown = (): UseDropdownReturnValue => {
   /**
    * Keydown event of combobox. Handles most of the keyboard accessibility of dropdown
    */
-  const onSelectKeydown: FormInputHandleOnKeyDownEvent = (e) => {
+  const onSelectKeydown = (e: { event: React.KeyboardEvent<HTMLInputElement> }): void => {
+    if (e.event.key === 'Tab' && rest.hasFooterAction) {
+      // When footer has Action Buttons, we ignore the blur event so that we can move focus to action item than bluring out of dropdown
+      setShouldIgnoreBlur(true);
+    }
+
     const actionType = getActionFromKey(e.event, isOpen);
+
     if (actionType) {
       performAction(actionType, e, {
         setIsOpen,
         onOptionChange,
         onComboType,
         selectCurrentOption: () => {
           selectOption(activeIndex);
+          if (rest.hasFooterAction) {
+            rest.selectInputRef.current?.focus();
+          }
+
+          const anchorLink = options[activeIndex]?.href;
+          if (anchorLink) {
+            window.location.href = anchorLink;
+            if (window.top) {
```

Comment:
why do we do this as well in addition to location.href?

Reply:
It is for scenarios where we're inside an iframe. E.g. storybook is inside iframe so it won't really work with storybook.